src/main/java/dinkplugin/message/DiscordMessageHandler.java

```diff
@@ -136,13 +156,57 @@ public void onFailure(@NotNull Call call, @NotNull IOException e) {
                 } else {
                     log.debug("Skipping retry attempts for failed webhook since max retries is not positive");
                 }
+
+                if (latch != null)
+                    latch.countDown();
             }
 
             @Override
             public void onResponse(@NotNull Call call, @NotNull Response response) {
                 log.trace("Successfully sent webhook message to {} after {} attempts", url, attempt + 1);
                 response.close();
+
+                if (latch != null)
+                    latch.countDown();
             }
         });
+
+        if (latch != null)
+            latch.await();
+    }
+
+    @VisibleForTesting
+    public boolean isAsync() {
+        return true;
+    }
```

Comment:
Why is it necessary for us to be able to control asyncness for the test notifications? does it have to do with rate limits or something?

Reply:
This was added for `TEST_WEBHOOK_URL` to function; when `sendMessage` is not blocking, junit terminates each test case too early (i.e., before it reliably has a chance to send the message to the webhook server) because the work is delegated to some threadpool. 

I can add a comment explaining this or we can consider alternative ways to get these tests working:
* Add `Thread#sleep` at the bottom of each test (bad imo)
* Switch `DiscordMessageHandler` to use `execute` instead of `enqueue` on okhttp requests, but perform this call on the passed `ScheduledExecutorService`. Then, in tests, we can define a special `Executor` that does the work on the main thread rather than creating a threadpool. This will effectively block the main thread without using `CountdownLatch`